java/google/registry/model/domain/rgp/GracePeriodStatus.java

```diff
@@ -101,4 +102,19 @@
   public String getXmlName() {
     return xmlName;
   }
+
+  /**
+   * Maps from xmlName to {@link GracePeriodStatus}.
+   *
+   * If no match is found for xmlName, null is returned.
+   */
+  @Nullable
+  public static GracePeriodStatus fromXmlName(String xmlName) {
+    for (GracePeriodStatus status : GracePeriodStatus.values()) {
```

Comment:
Would it be possible to replace this with a lookup in a static string-to-GracePeriodStatus map? This seems a little ugly. Ben suggests using the Guava Maps.uniqueIndex() static method, which takes a set of values and a function of those values (in this case, the xmlName strings), and returns a map.